assesspy/ci.py

```diff
@@ -43,69 +53,88 @@ def boot_ci(fun, nboot=100, alpha=0.05, **kwargs):
-        ["assessed", "sale_price"]
-    ).issubset(kwargs.keys()):
-        raise Exception(
-            "PRD function expects argurments 'assessed' and 'sale_price'."
-        )
-    else:
-        kwargs = tuple(kwargs.values())
+    # Take a random sample of input, with the same number of rows as input,
+    # with replacement
+    ests = pd.Series(index=range(nboot), dtype=float)
+    for i in range(nboot):
+        sample = df.sample(n=n, replace=True)
+        ests[i] = fun(sample.iloc[:, 0], sample.iloc[:, 1])
 
-    check_inputs(kwargs)  # Input checking and error handling
+    ci = (ests.quantile(alpha / 2), ests.quantile(1 - alpha / 2))
 
-    num_kwargs = len(kwargs)
-    kwargs = pd.DataFrame(kwargs).T
-    n = len(kwargs)
+    return ci
 
-    # Check that the input function returns a numeric vector
-    out = (
-        fun(kwargs.iloc[:, 0])
-        if num_kwargs < 2
-        else fun(kwargs.iloc[:, 0], kwargs.iloc[:, 1])
-    )
-    if not is_numeric_dtype(out):
-        raise Exception("Input function outputs non-numeric datatype.")
 
-    ests = []
+def cod_ci(
+    estimate: Union[list[int], list[float], pd.Series],
+    sale_price: Union[list[int], list[float], pd.Series],
+    nboot: int = 1000,
+    alpha: float = 0.05,
+) -> tuple[float, float]:
+    """
+    Calculate the non-parametric bootstrap confidence interval for COD.
 
-    # Take a random sample of input, with the same number of rows as input,
-    # with replacement.
-    for i in list(range(1, nboot)):
-        sample = kwargs.sample(n=n, replace=True)
-        if fun.__name__ == "cod" or num_kwargs == 1:
-            ests.append(fun(sample.iloc[:, 0]))
-        elif fun.__name__ == "prd":
-            ests.append(fun(sample.iloc[:, 0], sample.iloc[:, 1]))
-        else:
-            raise Exception(
-                "Input function should require 1 argument or be assesspy.prd."
-            )
-
-    ests = pd.Series(ests)
-
-    ci = [ests.quantile(alpha / 2), ests.quantile(1 - alpha / 2)]
+    See also:
+        :func:`boot_ci`
+    """
+    return boot_ci(
+        cod, estimate=estimate, sale_price=sale_price, nboot=nboot, alpha=alpha
+    )
 
-    return ci
 
+def prd_ci(
+    estimate: Union[list[int], list[float], pd.Series],
+    sale_price: Union[list[int], list[float], pd.Series],
+    nboot: int = 1000,
+    alpha: float = 0.05,
+) -> tuple[float, float]:
+    """
+    Calculate the non-parametric bootstrap confidence interval for PRD.
 
-# Formula specific bootstrapping functions
-def cod_ci(ratio, nboot=100, alpha=0.05):
-    return boot_ci(cod, ratio=ratio, nboot=nboot, alpha=alpha)
+    See also:
+        :func:`boot_ci`
+    """
+    return boot_ci(
+        prd, estimate=estimate, sale_price=sale_price, nboot=nboot, alpha=alpha
+    )
 
 
-def prd_ci(assessed, sale_price, nboot=100, alpha=0.05):
-    return boot_ci(
-        prd, assessed=assessed, sale_price=sale_price, nboot=nboot, alpha=alpha
+def prb_ci(
```

Comment:
The PRB CI was accessible before, but only via a dictionary value returned by `orb()`. This wasn't consistent with the other functions and also meant there was no `prb_ci()` function. I've added one here for API consistency even though it's largely redundant code with PRB.